criu/tty.c

```diff
@@ -1066,7 +1065,9 @@ static int pty_open_unpaired_slave(struct file_desc *d, struct tty_info *slave)
 		 * the process which keeps the master peer.
 		 */
 		if (root_item->sid != vpid(root_item)) {
-			if (root_item->pgid == vpid(root_item)) {
+			if (getppid() != tcgetpgrp(0)) {
+				pr_debug("Running in background, not setting foreground process");
+			} else if (root_item->pgid == vpid(root_item)) {
```

Comment:
Sorry for the delay with reply.

It not an easy change as it involves the state on checkpoint and the state on restore. So we are not restoring control group if restore runs in background.

Is it possible that we now leave some states where we should still be foreground? E.g. restoring in subshell like
```
$ ( ( java -XX:CRaCRestoreFrom= ... ) ) 
```
(no `&`, but parent probably not a foreground?)

Can restoring in background negatively impact java process which may assume foreground?

If the job moved to foreground (`java -XX:CRaCRestoreFrom= ... & ; fg), does it work correctly?